src/mrpro/data/IData.py

```diff
@@ -154,3 +160,167 @@ def from_dicom_folder(cls, foldername: str | Path, suffix: str | None = 'dcm') -
+        pattern_in = ['d' + str(i) for i in range(self.data.ndim)]
+        pattern_out = pattern_in.copy()
+        pattern_out[frame_dimension], pattern_out[-3] = pattern_out[-3], pattern_out[frame_dimension]
+        dcm_idata = self.rearrange(' '.join(pattern_in) + '->' + ' '.join(pattern_out))
+
+        # Metadata
+        file_meta = pydicom.dataset.FileMetaDataset()
+        file_meta.MediaStorageSOPClassUID = pydicom.uid.MRImageStorage
+        file_meta.MediaStorageSOPInstanceUID = pydicom.uid.generate_uid()
+        file_meta.TransferSyntaxUID = pydicom.uid.ExplicitVRLittleEndian
+
+        # Dataset
+        dataset = pydicom.Dataset()
+        dataset.file_meta = file_meta
+
+        dataset.PatientName = 'Unknown'
+        dataset.PatientID = 'Unknown'
+        dataset.PatientSex = 'O'
+        dataset.Modality = 'MR'
+        dataset.StudyDescription = 'MRpro'
+        import datetime
+
+        dataset.SeriesDate = datetime.datetime.now(datetime.timezone.utc).strftime('%Y%m%d')
+        dataset.SeriesTime = datetime.datetime.now(datetime.timezone.utc).strftime('%H%M%S.%f')
+        if series_description:
+            dataset.SeriesDescription = series_description
+            dataset.ProtocolName = series_description
+        dataset.SeriesInstanceUID = pydicom.uid.generate_uid()
+
+        dataset.PatientPosition = 'HFS'
+
+        for file_index, other in enumerate(np.ndindex(dcm_idata.shape[:-3])):
+            dcm_file_idata = dcm_idata[(*other, slice(None), slice(None), slice(None))]
+
+            dataset.MRAcquisitionType = mr_acquisition_type
+            dataset.PerFrameFunctionalGroupsSequence = pydicom.Sequence()
+
+            plane_position_sequence = pydicom.Sequence()
+            plane_position_sequence.append(Dataset())
+            plane_orientation_sequence = pydicom.Sequence()
+            plane_orientation_sequence.append(Dataset())
+            mr_echo_sequence = pydicom.Sequence()
+            mr_echo_sequence.append(Dataset())
+            pixel_measure_sequence = pydicom.Sequence()
+            pixel_measure_sequence.append(Dataset())
+            mr_timing_parameters_sequence = pydicom.Sequence()
+            mr_timing_parameters_sequence.append(Dataset())
+
+            for frame in range(number_of_frames):
+                dcm_frame_idata = dcm_file_idata[..., frame, :, :]
+                if dcm_frame_idata.header.shape.numel() != 1:
+                    raise ValueError('Only single image can be saved as a frame.')
+                directions = dcm_frame_idata.header.orientation[0].as_directions()
+                readout_direction = np.squeeze(torch.stack(directions[2].zyx[::-1]).numpy())
+                phase_direction = np.squeeze(torch.stack(directions[1].zyx[::-1]).numpy())
+                slice_direction = np.squeeze(torch.stack(directions[0].zyx[::-1]).numpy())
+                position = np.squeeze(torch.stack(dcm_frame_idata.header.position.zyx[::-1]).numpy())
```

Comment:
are these guaranteed to be the singleton and can be squeezed? what diemensions are squeezed?